add the auto_rescale_y on the GraphVisual model and make the depth of grid on GridVisual not relative to pixel width

fabien-colonnier opened the pull request:
Add the option to have rescale y in the append and in the update function of the GraphVisual.
It has been tested when both x and y are updated simultaneously. In update, it is not supporting 2 y axis yet. Not test was made to check it

optseb reviewed (changes requested):
Just a fix required for the graph example, either to reset it back to x^3 or change the labels to say that dataset 1 is a graph of y = x

optseb commented:
Oh, would it be a good idea and easy to throw an error if an update() was called with rescale_y  set true to avoid unexpected behaviour?

fabien-colonnier commented on examples/graph_incoming_data_rescale.cpp:
```diff
@@ -23,8 +23,8 @@ int main()
         morph::vvec<float> absc =  {-1.0, -.9, -.8, -.7, -.6, -.5, -.4, -.3, -.2, -.1, 0, .1, .2, .3, .4, .5, .6, .7, .8, .9, 1.0};
-        morph::vvec<float> data = absc.pow(3);
-        morph::vvec<float> data2 = absc.pow(5);
+        morph::vvec<float> data = 2.1f * absc.pow(1);
```
I corrected, I forgot to revert what I modify to validate it was working

fabien-colonnier commented:
I tested and corrected to have the full support of auto_rescale in the update function and added the auto_rescale_fit to fit the y axis. It is more complicated in the x axis as it is shared for different curves.

fabien-colonnier commented:
As I am writing this, I realize the auto_rescale_fit function doesn't handle multiple curves on one y axis. It will resize the axis according to the last dataset to be updated...

fabien-colonnier commented:
I added a comment for now, as It means rechecking all the datasets or storing the min and max for each one... which is tedious...

sebjameswml commented:
All looks good now.
